Subject: Linux-RT 2.6.27-RT
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
sched: only try to push a task on wakeup if it is migratable

From: Gregory Haskins <ghaskins@novell.com>

There is no sense in wasting time trying to push a task away that
cannot move anywhere else.  We gain no benefit from trying to push
other tasks at this point, so if the task being woken up is non
migratable, just skip the whole operation.  This reduces overhead
in the wakeup path for certain tasks.

Signed-off-by: Gregory Haskins <ghaskins@novell.com>
---

 kernel/sched_rt.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)


Index: linux-2.6.26-RT/kernel/sched_rt.c
===================================================================
--- linux-2.6.26-RT.orig/kernel/sched_rt.c
+++ linux-2.6.26-RT/kernel/sched_rt.c
@@ -1312,7 +1312,8 @@ static void task_wake_up_rt(struct rq *r
 {
 	if (!task_running(rq, p) &&
 	    !test_tsk_need_resched(rq->curr) &&
-	    rq->rt.overloaded) {
+	    rq->rt.overloaded &&
+	    p->rt.nr_cpus_allowed > 1) {
 		push_rt_tasks(rq);
 		schedstat_inc(rq, rto_wakeup);
 	}
